addons/event/models/event_mail.py

```diff
@@ -190,13 +173,51 @@ def execute(self):
                 })
         return True
 
+    def _execute_after_subscription(self, registration_limit=1000):
+        """ 'after_sub' scheduler management. It currently does two main things
+          * generate missing 'event.mail.registrations' which are scheduled
+            communication linked to registrations;
+          * launch registration-based communication, splitting in batches as
+            it may imply a lot of computation. When having more than given
+            limit to handle, schedule another call of cron to avoid having to
+            wait another cron interval check;
+        """
+        # fillup on subscription lines
+        if self.env.context.get('event_mail_registration_ids'):
+            new_registrations = self.env['event.registration'].search([
+                ('id', 'in', self.env.context['event_mail_registration_ids']),
+                ('event_id', '=', self.event_id.id),
+            ])
+        else:
+            new_registrations = self.event_id.registration_ids.filtered_domain(
+                [('state', 'not in', ('cancel', 'draft'))]
+            )
+        self._create_missing_mail_registrations(new_registrations)
+
+        # execute scheduler on registrations: limit number of registrations
+        # globally; if we have more next cron will handle them
+        todo = self.mail_registration_ids._filter_to_skip()
```

Comment:
To tie into my previous comment, "_filter_to_skip" will re-filter based on the registrations not being cancel or draft (except that it does it the other way around ^^).

Hopefully the cache will do some work here, but seems meh.
Would be preferably that "_filter_to_skip" becomes some kind of "_prepare_mail_registrations_domain" and then search in one go rather than fetching everything and then filtering. 

Especially since you would be able to limit the search to "registration_limit +1", to know if you have to re-launch or not, and save a lot of fetched records.